include/xrpl/beast/utility/instrumentation.h

```diff
@@ -0,0 +1,58 @@
+//------------------------------------------------------------------------------
+/*
+This file is part of rippled: https://github.com/ripple/rippled
+Copyright (c) 2024 Ripple Labs Inc.
+
+Permission to use, copy, modify, and/or distribute this software for any
+purpose  with  or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef BEAST_UTILITY_INSTRUMENTATION_H_INCLUDED
+#define BEAST_UTILITY_INSTRUMENTATION_H_INCLUDED
+
+#include <cassert>
+
+#ifdef ENABLE_VOIDSTAR
+#ifdef NDEBUG
+#error "Antithesis instrumentation requires Debug build"
+#endif
+#include <antithesis_sdk.h>
+#else
+#define ALWAYS(cond, name, ...) assert((name) && (cond))
+#define ALWAYS_OR_UNREACHABLE(cond, name, ...) assert((name) && (cond))
+#define SOMETIMES(cond, name, ...)
+#define REACHABLE(name, ...)
+#define UNREACHABLE(name, ...) assert((name) && false)
+#endif
+
+#define ASSERT ALWAYS_OR_UNREACHABLE
+
+// How to use the above macros:
+//
+// ALWAYS if it must be reached during fuzzing
+// ASSERT if you expect that it might, or might not, be reached during fuzzing
+// REACHABLE if it must be reached during fuzzing
```

Comment:
The `ALWAYS` and `REACHABLE` have the same usage description, but must have different definitions. Can you clarify their differences here?

Reply:
the difference is that one takes `cond` parameter and the other does not - I think this is visible enough ?

Reply:
It is visible, but requires the reader to look at both the define statements and the comments to figure that out. I figured the reachable description could be something like `REACHABLE if it must be reached during fuzzing provided the specified condition holds`.

Reply:
It's not "provided" - this would imply that, if the condition does not hold, the location may not be reached. I will figure out better explanation.